cadc-util/src/main/java/ca/nrc/cadc/auth/DelegationToken.java

```diff
@@ -190,40 +266,70 @@ public static DelegationToken parse(String text, String requestURI, ScopeValidat
     {
         String[] fields = text.split(FIELD_DELIM);
         String userid = null;
+        Set<Principal> principalSet = new HashSet<>();
         String proxyUser = null;
         Date expirytime = null;
         URI scope = null;
         String signature = null;
+        List<String> domains = new ArrayList<>();
         try
         {
             for (String field : fields)
             {
+
                 String key = field.substring(0, field.indexOf(VALUE_DELIM));
                 String value = field.substring(field.indexOf(VALUE_DELIM) + 1);
-                if (key.equalsIgnoreCase("userid"))
+                log.debug("key = value: " + key + "=" + value);
+
+                if (key.equalsIgnoreCase(IdentityType.USERID.getValue()))
                 {
                     userid = value;
                 }
-                if (key.equalsIgnoreCase("proxyuser"))
+                if (key.equalsIgnoreCase(PROXY_LABEL))
```

Comment:
sequence of if should be "else if"?

Reply:
switch on strings is available as of Java 7. At the very least I inserted the elsif to reduce the amount of wasted calls.